lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,423 @@
+      command_name _('update-attributes')
+      resource :compute_attributes
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface-id', 'INTERFACE_ID', _('Interface id'),
+             :format => HammerCLI::Options::Normalizers::Number.new
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--volume-id', 'VOLUME_ID', _('Volume id'),
+             :format => HammerCLI::Options::Normalizers::Number.new
+
+      extend_help do |h|
+        h.section _('Provider specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.compute_profile_create_help)
+                provider.compute_profile_create_help(h)
+              else
+                provider.host_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name ).required
+        validator.any(:option_compute_resource_id, :option_compute_resource_name).required
+        validator.any(:option_interface, :option_volume, :option_compute_attributes).required
+
+      end
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        interface_attr =   params['compute_attribute']['vm_attrs']['interfaces_attributes'] || {}
+        volume_attr =  params['compute_attribute']['vm_attrs']['volums_attributes'] || {}
+        vm_attr =  params['compute_attribute']['vm_attrs'] || {}
+        params['id'] =  params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs'] = params['compute_attribute']['vm_attrs'].merge(vm_attr.merge(option_compute_attributes || {} ))
+
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({option_interface_id => option_interface})
+        end
+        if option_volume
+          unless option_volume_id
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = volume_attr.merge({option_volume_id => option_volume})
+          end
+        end
+        params
+      end
+      success_message _('Compute profile attributes updated')
+      failure_message _('Could not update the compute profile attributes')
+
+      build_options :without => :id
+    end
+
+    class AddInterface < HammerCLIForeman::UpdateCommand
```

Comment:
Maybe it should inherit `CreateCommand` instead of `UpdateCommand`?

Also it doesn't seem to work. I tried `hammer compute-profile add-interface --compute-resorce tstracho-laptop --compute-profile 1-Small --set-values type=network,bridge=br1` and got ```Could not create interface: Error: undefined method `[]' for nil:NilClass```. Does it work for you?

Reply:
bridge is not allowed attribute for any of the compute resources 
check hammer compute-profile add-interface --help
 and I can't reproduce this error at all..  are the compute profile and compute resource name are exists?
what is the compute resource your testing on? ovirt?

Reply:
the relation - can each compute profile have only one compute attribute per compute resource?
^^ you  can have only one compute attribute, but inside the compute attribute you can have multiple volumens and interfaces.